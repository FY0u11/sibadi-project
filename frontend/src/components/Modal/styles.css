.my-container {
  position: fixed;
  display: flex;
  align-items: center;
  justify-content: center;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background: rgba(0, 0, 0, 0.5);
  z-index: 1031;
}

.my-modal {
  width: 400px;
  height: 200px;
  padding: 1rem;
  background: #fff;
  border-radius: 10px;
}
