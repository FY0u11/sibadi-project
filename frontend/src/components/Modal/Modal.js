import React from 'react'

import './styles.css'

const Modal = ({ modal, closeModal }) => {
  return (
    <div
      className="my-container"
      onMouseDown={event => event.target.className === 'my-container' && closeModal()}
    >
      <div className="my-modal">
        <button className="ml-2 mb-1 close" onClick={closeModal}>
          &times;
        </button>
        <h2>{modal.title}</h2>
        {modal.body}
      </div>
    </div>
  )
}

export default Modal
