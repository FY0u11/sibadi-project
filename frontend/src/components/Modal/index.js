import React from 'react'
import { connect } from 'react-redux'

import { closeModal } from '../../store/components/modal/actions'

import Modal from './Modal'

const ModalContainer = ({ modal, closeModal }) => {
  return modal ? <Modal modal={modal} closeModal={closeModal} /> : null
}

const putStateToProps = state => ({ modal: state.modal.modal })
const putActionsToProps = { closeModal }

export default connect(putStateToProps, putActionsToProps)(ModalContainer)
