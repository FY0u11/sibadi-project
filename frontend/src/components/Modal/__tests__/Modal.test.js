import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import Modal from '../Modal'

afterEach(cleanup)

const modal = { title: 'Modal title', body: 'Modal body' }

describe('Testing Modal component', () => {
  it('It should be properly rendered', () => {
    const { asFragment } = render(<Modal modal={modal} />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('It should trigger on mouse down event', () => {
    const callback = jest.fn()
    const { container, getByText } = render(<Modal modal={modal} closeModal={callback} />)
    const modalContainer = container.children.item(0)
    const closeButton = getByText('Ã—')
    expect(modalContainer).toHaveClass('my-container')
    expect(closeButton).toHaveClass('close')
    fireEvent.mouseDown(modalContainer)
    fireEvent.click(closeButton)
    expect(callback).toBeCalledTimes(2)
  })
})
