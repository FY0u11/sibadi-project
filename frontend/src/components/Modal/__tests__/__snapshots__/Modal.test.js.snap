// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Modal component It should be properly rendered 1`] = `
<DocumentFragment>
  <div
    class="my-container"
  >
    <div
      class="my-modal"
    >
      <button
        class="ml-2 mb-1 close"
      >
        Ã—
      </button>
      <h2>
        Modal title
      </h2>
      Modal body
    </div>
  </div>
</DocumentFragment>
`;
